api/app/Repositories/UserRepository.php

```diff
@@ -15,13 +15,50 @@ class UserRepository extends BaseRepository
     protected $login_token_ttl = 1440;
 
     /**
+     * Cache repository.
+     *
+     * @var Cache
+     */
+    protected $cache;
+
+    /**
+     * Assign dependencies.
+     *
+     * @param  Connection  $connection
+     * @param  Cache       $cache
+     * @return void
+     */
+    public function __construct(Connection $connection, Cache $cache)
+    {
+        parent::__construct($connection);
+
+        $this->cache = $cache;
+    }
+
+    /**
      * Model name.
      *
-     * @return string
+     * @return User
      */
     protected function model()
     {
-        return new User();
+        return new User;
+    }
+
+    /**
+     * Check if a user exists.
+     *
+     * @param  string  $id
+     * @return bool
+     */
+    public function exists($id)
```

Comment:
This method isn't user specific and is probably useful elsewhere, so can you move it to the `BaseRepository` please?